Allow AnnularMeshGenerator to create constant-area elements. (#20423)
idaholab/moose: framework/doc/content/source/meshgenerators/AnnularMeshGenerator.md

The number of elements in the radial direction and the angular direction may be specified. By default, the mesh spacing is uniform in the radial direction, but the following options are available for more control over the radial meshing:
- A growth factor that controls the element size in the radial direction may be specified.  In this case, the radial element size in the innermost ring is multiplied by this factor for each concentric ring of elements, moving from the inner to the outer radius. If the growth factor is positive, the element radial dimension increases with increasing radial position, while if the growth factor is negative, the element radial dimension decreases with increasing radial position.
- A list of values that define the radial positions of the rings of interior nodes can be specified. This allows for direct control of the element size for every one of the concentric rings. In this case, the number of radial elements is not specified, because it is inferred from the length of that list.

> - An on/off flag as to whether to enforce equal areas among all the elements. When true, this parameter will automatically determine the radial element spacing so that each element has the same area.

Reviewer: ```suggestion
- A flag that enforces equal areas among all of the elements. When true, this parameter will automatically determine the radial element spacing so that each element has the same area.
```
